IntervalTree/IntervalTreeNode.cs

```diff
@@ -171,30 +173,8 @@ public IEnumerable<TValue> Query(TKey from, TKey to)
             return fromComp;
         }
 
-        public TKey Max
-        {
-            get
-            {
-                if (rightNode != null)
-                    return rightNode.Max;
-                else if (items != null)
-                    return items.Max(i => i.To);
-                else
-                    return default(TKey);
-            }
-        }
+        public TKey Max { get; private set; } = default(TKey);
```

Comment:
Why the private setter?

Reply:
I would also prefer to investigate first, if we can fix the issue directly in the getter.

Reply:
To elaborate a bit more on my comment, I think this should be sufficient.
```cs
public TKey Min { get; }
public TKey Max { get; }
```

Reply:
I admit that the private setter is redundant, as is the initialization with default.

But why using a computed property instead of a value property? 
If the getter calculates the Max value with each access, this will be rather expensive, expecially using large amounts of data. The property can be set by the constructor without further calculations. I think one can spare the memory for two pointers/worst case structs?

Reply:
If the value never changes after initial construction, I agree, that we can just store the value in a field. I just don't know (by heart) why it was a computed property in the first place (I assumed because it could change, as new nodes are added to the graph).

Reply:
adding a new node would invalidate the tree, resulting in a rebuild, which essentially creates a new root node. So i don't see a way this could change.